soroban-sdk/src/crypto.rs

```diff
@@ -71,15 +71,15 @@ impl<const N: usize> IntoVal<Env, BytesN<N>> for Hash<N> {
     }
 }
 
-impl<const N: usize> Into<Bytes> for Hash<N> {
-    fn into(self) -> Bytes {
-        self.0.into()
+impl<const N: usize> From<Hash<N>> for Bytes {
+    fn from(v: Hash<N>) -> Self {
+        v.0.into()
     }
 }
 
-impl<const N: usize> Into<BytesN<N>> for Hash<N> {
-    fn into(self) -> BytesN<N> {
-        self.0
+impl<const N: usize> From<Hash<N>> for BytesN<N> {
+    fn from(v: Hash<N>) -> Self {
+        v.0
```

Comment:
From impls result in an automatic Into implementation anyway, but provide the additional from functions on the Bytes type.